### PR TITLE
Make use of OpenMLS' `PublicGroup`

### PR DESCRIPTION
**Pull request by kkohbrok**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,11 +11,27 @@ features = "0.10.0"
 openmls_traits = "0.1.0"
 serde = { version = "1.0.147", features = ["derive"] }
 tls_codec = { version = "0.2.0", features = ["derive", "serde", "mls"] }
+openmls = { path = "../../openmls/openmls/openmls" }
+openmls_rust_crypto = { version = "0.1.0", path = "../../openmls/openmls/openmls_rust_crypto" }
```
Comment (raphaelrobert) [inline Cargo.toml:15]: These look like local paths

File: Cargo.toml [hunk 2]
```diff
@@ -11,11 +11,27 @@ features = "0.10.0"
...
 git = "https://github.com/RustCrypto/formats.git"
 
 [patch.crates-io.openmls_traits]
-git = "https://github.com/openmls/openmls.git"
\ No newline at end of file
+git = "https://github.com/openmls/openmls.git"
+
+##[patch.crates-io.openmls]
+##git = "https://github.com/openmls/openmls.git"
+##branch = "konrad/public_group_api"
```
Comment (raphaelrobert) [inline Cargo.toml:28]: Remove?

File: src/group/mod.rs [hunk 3]
```diff
@@ -1,71 +1,175 @@
...
 
 impl Group {
-    pub fn merge_staged_commit(&mut self, _staged_commit: StagedCommit) {}
-
-    pub fn public_tree() {}
+    /// Create a new group state with the group consisting of the creator's
+    /// leaf.
+    pub fn new(
+        verifiable_group_info: VerifiableGroupInfo,
+        leaf_node: LeafNode,
```
Comment (raphaelrobert) [inline src/group/mod.rs:35]: In theory, this could also be a `KeyPackage`, no? But whatever works for now.

File: src/group/mod.rs [hunk 4]
```diff
@@ -1,71 +1,175 @@
...
+            group_info,
+            public_group,
+            backend,
+        })
+    }
 
-    pub fn group_info(&self) -> &GroupInfo {
-        &self.group_info
+    fn backend(&self) -> &OpenMlsRustCrypto {
+        &self.backend
```
Comment (raphaelrobert) [inline src/group/mod.rs:53]: The name is a tad confusing here

File: src/group/mod.rs [hunk 5]
```diff
@@ -1,71 +1,175 @@
...
+                }
+            }
+        }
+    }
 
-#[derive(
-    Debug, PartialEq, Eq, Clone, Serialize, Deserialize, TlsDeserialize, TlsSerialize, TlsSize,
-)]
-pub struct GroupId;
+    pub fn accept_message(&mut self, processed_assisted_message: ProcessedAssistedMessage) {
```
Comment (raphaelrobert) [inline src/group/mod.rs:110]: Can you add a comment on what this does?

**Review by raphaelrobert — CHANGES_REQUESTED** [review]
Thanks! Just some nits

File: src/group/mod.rs (hunk 3 above)
Comment (kkohbrok) [reply src/group/mod.rs:35]: Sure, we will have to change the API anyway when we start using mls-assist properly.

File: src/group/mod.rs (hunk 4 above)
Comment (kkohbrok) [reply src/group/mod.rs:53]: Is it? Is there a name you would prefer?

File: src/group/mod.rs (hunk 5 above)
Comment (kkohbrok) [reply src/group/mod.rs:112]: Will do. I'm also not super happy with the name, so if you have a suggestion, I'm all ears.

File: Cargo.toml (hunk 1 above)
Comment (kkohbrok) [reply Cargo.toml]: Sorry about that. OpenMLS needs a few changes to its interface before we can use it, so I used a local version for development. The changes are now on a branch called `konrad/public_group_interface`.

**Review by kkohbrok — COMMENTED** [review]
Thanks for the review!

File: src/group/mod.rs (hunk 4 above)
Comment (raphaelrobert) [reply src/group/mod.rs:53]: Backend is already used elsewhere, maybe something more precise like crypto provider

**Review by raphaelrobert — APPROVED** [review]
Just a nit, but I don't have a strong opinion

**Pull request merged**